src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java

```
@@ -0,0 +1,134 @@
+
+    MQConnectionBuilder connectionBuilder;
+    MQConfiguration config;
+    MQTopic topic = null;
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+
+            log.info("Message received at ibm-mq connector");
+            connectionBuilder = new MQConnectionBuilder(messageContext);
+            config = connectionBuilder.getConfig();
+
+            SOAPEnvelope soapEnvelope = messageContext.getEnvelope();
+            OMElement getElement = soapEnvelope.getBody().getFirstElement();
+            String queueMessage = "";
+
+            if (getElement != null) {
+                queueMessage = getElement.toString();
+            }
+
+            MQMessage mqMessage = buildMQMessage(config, queueMessage);
+
+            if (config.getTopicName() != null) {
+                topic = setTopic(connectionBuilder, config);
+            }
+
+            if (topic == null || mqMessage == null) {
+                log.error("Error in publishing message");
+            } else {
+                log.info("topic initialized");
+                topic.put(mqMessage);
+                log.info("Message successfully placed at " + config.getTopicName() + "topic");
+            }
+
+            connectionBuilder.closeConnection();
+
+        } catch (Exception e) {
+            log.error("Problem in publishing to message" + e);
+            throw new ConnectException(e);
+        }
+    }
+
+    /**
+     * Initialize topic
+     */
+    MQTopic setTopic(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQTopic publisher;
+        try {
+            publisher = manager.accessTopic(config.getTopicString(), config.getTopicName(),
+                    CMQC.MQTOPIC_OPEN_AS_PUBLICATION, CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating topic" + e);
+            return null;
```

Returning null and handle error as generic, we will be loosing information why this got failed. It s better if we could throw this exception to higher layer and handle over there.